### DISCUSSION TITLE
Website get detected?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [ultrafunkamsterdam/undetected-chromedriver#1771] [General]
Hello - i try to get the following site using the following code:

```
import undetected_chromedriver as uc
driver = uc.Chrome(version_main=122)
driver.get('https://ballzy.eu/en/men/sport/shoes') 
input("Press!")
```

But when i run this code it seems the driver is still detected and i get the cloudflare challenge.
Why is this not working using the above code?

**Comment by mdmintz** [comment]
You can try [SeleniumBase](https://github.com/seleniumbase/SeleniumBase) UC Mode. The algorithm is 90% undetected-chromedriver's algorithm, but with some changes around naming, options, customizations, default timing, and new methods. (Some changes have open pull requests in this repo.)

```python
from seleniumbase import Driver

driver = Driver(uc=True)
driver.uc_open_with_reconnect("https://ballzy.eu/en/men/sport/shoes", 5)

breakpoint()

driver.quit()
```

I didn't see any CAPTCHAs when I ran that. Type `c` and press Enter to continue from the `breakpoint()`.

**Comment by Rapid1898-code** [comment]
ok i understand - the reason why i am asking that one of my clients has a problem with selenium base on his computer
and so i tread to simply use this library as an alternative
but when this is not working with this url using the undetected-chromedriver library - i have to stick to seleniumbase unfortunately

**Reply by mdmintz** [reply]
The code I showed is using SeleniumBase.

**Reply by Rapid1898-code** [reply]
yes i understand - but i would need an alternative solution for the url using this module and not seleniumbase
(cause some client of mine has problems running the solution with seleniumbase on his computer)

**Reply by mdmintz** [reply]
Use the `SB(uc=True, xvfb=True)` format. It'll work in any environment:

```python
from seleniumbase import SB

with SB(uc=True, xvfb=True) as sb:
    sb.driver.uc_open_with_reconnect("https://ballzy.eu/en/men/sport/shoes", 5)

    breakpoint()
```

**Reply by Rapid1898-code** [reply]
ok - thats interesting - in my actual solution i have this parameters when working:

`with SB(uc=True, headless=True) as sb: `

I will try this with the ones you have suggested.

This is btw. the full error-log i currently have from the client:

```
Working for page https://ballzy.eu/en/men/sport/shoes

Warning: uc_driver not found. Getting it now:

*** chromedriver to download = 122.0.6261.94 (Latest Stable)

Downloading chromedriver-win64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/122.0.6261.94/win64/chromedriver-win64.zip ...
Download Complete!

*** chromedriver to download = 122.0.6261.94 (Latest Stable)
Downloading chromedriver-win64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/122.0.6261.94/win64/chromedriver-win64.zip ...
Download Complete!

DevTools listening on ws://127.0.0.1:64766/devtools/browser/7ff08603-033f-44f8-a151-44ecf8c8fb92
Traceback (most recent call last):
  File "collBallzy4.py", line 186, in <module>
AttributeError: 'WebDriver' object has no attribute 'uc_open_with_reconnect'
[59388] Failed to execute script 'collBallzy4' due to unhandled exception!
```

As you can see at the end it is crashing.

**Reply by mdmintz** [reply]
That's using an older version of `seleniumbase`. You need to upgrade to the latest version. It's running into the same issue that `undetected-chromedriver` users ran into here: https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/1743 Since then, both `undetected-chromedriver` and `seleniumbase` shipped newer versions with the fix. (The Chromium Team changed the download location.)

Also, the `driver` only has `uc_open_with_reconnect()` when running in UC Mode (`uc=True`).

**Comment by Rapid1898-code** [comment]
Hello - i tested it now with the following code-snippet -

```
from seleniumbase import SB
from bs4 import BeautifulSoup

with SB(uc=True, xvfb=True) as sb:
  sb.driver.uc_open_with_reconnect("https://ballzy.eu/en/men/sport/shoes", 5)
  soup = BeautifulSoup (sb.driver.page_source, 'lxml')         
  tmpDIV = soup.find("div", {"data-block-id": "ku_landing_result_items"})
  tmpDIV = tmpDIV.find_all("li")  
  print(f"{len(tmpDIV)} elements found")
```

When i run the python-file normally it seems to work with this output:

```
(seleniumbase) C:\DEV\Fiverr\ORDER\stefamn_jan669_jankore_janxx2\Ballzy>python test1.py

Warning: uc_driver update needed. Getting it now:

*** chromedriver to download = 122.0.6261.94 (Latest Stable) 

Downloading chromedriver-win64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/122.0.6261.94/win64/chromedriver-win64.zip ...
Download Complete!

12 elements found
```

But when i bundle the code to a windows executable with this command

`pyinstaller --onefile test1.py`

I get this error / log:

```
C:\DEV\Fiverr\ORDER\stefamn_jan669_jankore_janxx2\Ballzy>test1.exe

Warning: uc_driver not found. Getting it now:

*** chromedriver to download = 122.0.6261.94 (Latest Stable)

Downloading chromedriver-win64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/122.0.6261.94/win64/chromedriver-win64.zip ...
Download Complete!

Traceback (most recent call last):
  File "test1.py", line 5, in <module>
    sb.driver.uc_open_with_reconnect("https://ballzy.eu/en/men/sport/shoes", 5)
AttributeError: 'WebDriver' object has no attribute 'uc_open_with_reconnect'
[7776] Failed to execute script 'test1' due to unhandled exception!
```

Why is this not working with the bundled exe-file?
Or can i create and use the undetected-driver somehow else?
